src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -0,0 +1,72 @@
+package com.aws.greengrass.logmanager.model;
+
+import com.aws.greengrass.util.Utils;
+import lombok.Getter;
+
+import java.io.File;
+import java.net.URI;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static com.aws.greengrass.logmanager.LogManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG;
+
+public final class LogFileGroup {
+    @Getter
+    List<LogFile> logFiles;
+
+    private LogFileGroup(List<LogFile> files) {
+        this.logFiles = files;
+    }
+
+    /**
+     * Create a list of Logfiles that are sorted based on lastModified time.
+     * @param filePattern the fileNameRegex used for each component to recognize its log files.
+     * @param path the directory path of the log files of component.
+     * @param lastUpdated the saved updated time of the last uploaded log of a component.
+     * @return list of logFile.
+     */
+    public static LogFileGroup create(Pattern filePattern, URI path, Instant lastUpdated) {
+        File folder = new File(path);
+
+        if (!folder.isDirectory()) {
+            throw new RuntimeException("Must be a folder.");
```

Comment:
Maybe don't use a RuntimeException and have a custom checked exception like `InvalidLogGroupException`